fastlane/spec/helper/tool_name_formatting_helper_spec.rb

```diff
@@ -0,0 +1,18 @@
+require_relative '../../helper/tool_name_formatting_helper.rb'
+
+describe Fastlane::Helper::ToolNameFormattingHelper do
+  before(:each) do
+    content = File.read("fastlane/spec/fixtures/fastfiles/tool_name_formatting.txt")
+    @helper = Fastlane::Helper::ToolNameFormattingHelper.new(content: content, path: "CONTRIBUTING.md")
+  end
+
+  describe "when parsing fixture file" do
+    it "should match array of expected errors" do
+      expected_errors = [
+        "fastlane tools have to be formatted in lowercase: fastlane in 'CONTRIBUTING.md:19': _FastLane_ makes code signing management easy. ❌",
+        "fastlane tools have to be formatted in lowercase: fastlane in 'CONTRIBUTING.md:23': A sentence that contains a _FastLane_ keyword, but also an env var: `FASTLANE_SKIP_CHANGELOG`. ❌"
+      ]
+      expect(@helper.find_tool_name_formatting_errors).to match_array(expected_errors)
```

Comment:
Yep exactly what I had in mind. Because this not only ensures that what we expect to be reported as errors are indeed reported as errors, but also checks that what we don't expect to be reported as errors and expect to be valid exceptions… isn't accidentally reported as errors either.

Btw, now that we have this implemented as a `spec`, I think it's worth removing the ❌  and ✅  at the end of each lines of your fixtures now, as they are not a needed hack anymore, and could accidentally mislead the thing being tested (e.g. imagine if our implementation of `find_tool_name_formatting_errors` didn't check if the lines of `CONTRIBUTING.md` _started_  with "- ❌", but instead our implementation tested if `include?("❌")`… we don't want those emojis at end of line to mud the waters here.

Reply:
I thought about removing them but ended up leaving them (hence why I reworded the header of the fixture file), because I think it's still useful when we need to update this fixture in the future. I think the header explains all that so there's no "accidentally misleading" I think 🤔 unless devs don't read it. But you have a good point nonetheless, I'll remove them